packages/valory/skills/abstract_round_abci/models.py

```diff
@@ -101,11 +101,20 @@ def _ensure(cls, key: str, kwargs: Dict) -> Any:
 class SharedState(Model):
     """Keep the current shared state of the skill."""
 
-    def __init__(self, *args: Any, abci_app_cls: Type[AbciApp], **kwargs: Any) -> None:
+    def __init__(
+        self,
+        *args: Any,
+        abci_app_cls: Type[AbciApp],
+        skill_context: SkillContext,
+        **kwargs: Any,
+    ) -> None:
         """Initialize the state."""
         self.abci_app_cls = self._process_abci_app_cls(abci_app_cls)
+        self.abci_app_cls._is_abstract = (
+            skill_context._skill.configuration.is_abstract_component  # pylint: disable=protected-access
+        )
```

Comment:
@angrybayblade I'd suggest that we create a property for `is_abstract_component` at `open-aea`'s `SkillContext`.

Reply:
the problem is that `is_abstract` should not be modified arbitrarily, but only propertly by `SharedState`, hence it was a protected class attributes.

We could have a public access method though.

Reply:
#1093.